### PR TITLE
vim: add a utility plugin, a few themes and bump the rest

### PR DESCRIPTION
**Pull request by lucasew**
###### Description of changes
Basically:
```
cd pkgs/applications/editors/vim/plugins
./update.py add josegamez82/starrynight https://github.com/yorickpeterse/vim-paper https://github.com/ewilazarus/preto https://github.com/lucasew/yescapsquit.vim
./update.py
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 245275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vimPlugins.sg-nvim</li>
  </ul>
</details>
<details>
  <summary>112 packages built:</summary>
  <ul>
    <li>astroid</li>
    <li>spacevim</li>
    <li>vim-full</li>
    <li>vimPlugins.ChatGPT-nvim</li>
    <li>vimPlugins.Coqtail</li>
    <li>vimPlugins.LazyVim</li>
    <li>vimPlugins.LeaderF</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.ale</li>
    <li>vimPlugins.barbecue-nvim</li>
    <li>vimPlugins.base46</li>
    <li>vimPlugins.bufferline-nvim</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.clangd_extensions-nvim</li>
    <li>vimPlugins.cmp-fuzzy-buffer</li>
    <li>vimPlugins.cmp-fuzzy-path</li>
    <li>vimPlugins.cmp-neosnippet</li>
    <li>vimPlugins.codeium-vim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.context_filetype-vim</li>
    <li>vimPlugins.coq-artifacts</li>
    <li>vimPlugins.coq-thirdparty</li>
    <li>vimPlugins.coq_nvim</li>
    <li>vimPlugins.flash-nvim</li>
    <li>vimPlugins.flatten-nvim</li>
    <li>vimPlugins.flit-nvim</li>
    <li>vimPlugins.flutter-tools-nvim</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.fuzzy-nvim</li>
    <li>vimPlugins.glance-nvim</li>
    <li>vimPlugins.haskell-tools-nvim</li>
    <li>vimPlugins.heirline-nvim</li>
    <li>vimPlugins.lazy-nvim</li>
    <li>vimPlugins.lazygit-nvim</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.leap-nvim</li>
    <li>vimPlugins.lspsaga-nvim-original</li>
    <li>vimPlugins.mason-lspconfig-nvim</li>
    <li>vimPlugins.mason-nvim</li>
    <li>vimPlugins.mason-tool-installer-nvim</li>
    <li>vimPlugins.mini-nvim</li>
    <li>vimPlugins.minimap-vim</li>
    <li>vimPlugins.ncm2-neosnippet</li>
    <li>vimPlugins.neodev-nvim</li>
    <li>vimPlugins.neogit</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neosnippet-vim</li>
    <li>vimPlugins.neotest</li>
    <li>vimPlugins.neotest-haskell</li>
    <li>vimPlugins.nix-develop-nvim</li>
    <li>vimPlugins.noice-nvim</li>
    <li>vimPlugins.nvchad</li>
    <li>vimPlugins.nvchad-extensions</li>
    <li>vimPlugins.nvchad-ui</li>
    <li>vimPlugins.nvim-base16</li>
    <li>vimPlugins.nvim-cokeline</li>
    <li>vimPlugins.nvim-gdb</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-navic</li>
    <li>vimPlugins.nvim-pqf</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-spectre</li>
    <li>vimPlugins.nvim-spider</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.elvish</li>
    <li>vimPlugins.nvim-treesitter-parsers.erlang</li>
    <li>vimPlugins.nvim-treesitter-parsers.fortran</li>
    <li>vimPlugins.nvim-treesitter-parsers.markdown</li>
    <li>vimPlugins.nvim-treesitter-parsers.markdown_inline</li>
    <li>vimPlugins.nvim-treesitter-parsers.matlab</li>
    <li>vimPlugins.nvim-treesitter-parsers.ocaml</li>
    <li>vimPlugins.nvim-treesitter-parsers.ocaml_interface</li>
    <li>vimPlugins.nvim-treesitter-parsers.robot</li>
    <li>vimPlugins.nvim-treesitter-parsers.tsx</li>
    <li>vimPlugins.nvim-treesitter-parsers.typescript</li>
    <li>vimPlugins.nvim-treesitter-parsers.wing</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.onedarkpro-nvim</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.preto</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.rose-pine</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.starrynight</li>
    <li>vimPlugins.tabline-nvim</li>
    <li>vimPlugins.telescope-cheat-nvim</li>
    <li>vimPlugins.telescope-frecency-nvim</li>
    <li>vimPlugins.telescope-fzf-native-nvim</li>
    <li>vimPlugins.telescope-fzf-writer-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-media-files-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.telescope-symbols-nvim</li>
    <li>vimPlugins.telescope-undo-nvim</li>
    <li>vimPlugins.telescope-z-nvim</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.trouble-nvim</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vifm-vim</li>
    <li>vimPlugins.vim-just</li>
    <li>vimPlugins.vim-nix</li>
    <li>vimPlugins.vim-paper</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vimtex</li>
    <li>vimPlugins.wildfire-vim</li>
    <li>vimPlugins.yescapsquit-vim</li>
    <li>vimPlugins.zig-vim</li>
    <li>vimacs</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 245275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vimPlugins.sg-nvim</li>
  </ul>
</details>
<details>
  <summary>112 packages built:</summary>
  <ul>
    <li>astroid</li>
    <li>spacevim</li>
    <li>vim-full</li>
    <li>vimPlugins.ChatGPT-nvim</li>
    <li>vimPlugins.Coqtail</li>
    <li>vimPlugins.LazyVim</li>
    <li>vimPlugins.LeaderF</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.ale</li>
    <li>vimPlugins.barbecue-nvim</li>
    <li>vimPlugins.base46</li>
    <li>vimPlugins.bufferline-nvim</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.clangd_extensions-nvim</li>
    <li>vimPlugins.cmp-fuzzy-buffer</li>
    <li>vimPlugins.cmp-fuzzy-path</li>
    <li>vimPlugins.cmp-neosnippet</li>
    <li>vimPlugins.codeium-vim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.context_filetype-vim</li>
    <li>vimPlugins.coq-artifacts</li>
    <li>vimPlugins.coq-thirdparty</li>
    <li>vimPlugins.coq_nvim</li>
    <li>vimPlugins.flash-nvim</li>
    <li>vimPlugins.flatten-nvim</li>
    <li>vimPlugins.flit-nvim</li>
    <li>vimPlugins.flutter-tools-nvim</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.fuzzy-nvim</li>
    <li>vimPlugins.glance-nvim</li>
    <li>vimPlugins.haskell-tools-nvim</li>
    <li>vimPlugins.heirline-nvim</li>
    <li>vimPlugins.lazy-nvim</li>
    <li>vimPlugins.lazygit-nvim</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.leap-nvim</li>
    <li>vimPlugins.lspsaga-nvim-original</li>
    <li>vimPlugins.mason-lspconfig-nvim</li>
    <li>vimPlugins.mason-nvim</li>
    <li>vimPlugins.mason-tool-installer-nvim</li>
    <li>vimPlugins.mini-nvim</li>
    <li>vimPlugins.minimap-vim</li>
    <li>vimPlugins.ncm2-neosnippet</li>
    <li>vimPlugins.neodev-nvim</li>
    <li>vimPlugins.neogit</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.neosnippet-vim</li>
    <li>vimPlugins.neotest</li>
    <li>vimPlugins.neotest-haskell</li>
    <li>vimPlugins.nix-develop-nvim</li>
    <li>vimPlugins.noice-nvim</li>
    <li>vimPlugins.nvchad</li>
    <li>vimPlugins.nvchad-extensions</li>
    <li>vimPlugins.nvchad-ui</li>
    <li>vimPlugins.nvim-base16</li>
    <li>vimPlugins.nvim-cokeline</li>
    <li>vimPlugins.nvim-gdb</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-navic</li>
    <li>vimPlugins.nvim-pqf</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-spectre</li>
    <li>vimPlugins.nvim-spider</li>
    <li>vimPlugins.nvim-tree-lua</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.elvish</li>
    <li>vimPlugins.nvim-treesitter-parsers.erlang</li>
    <li>vimPlugins.nvim-treesitter-parsers.fortran</li>
    <li>vimPlugins.nvim-treesitter-parsers.markdown</li>
    <li>vimPlugins.nvim-treesitter-parsers.markdown_inline</li>
    <li>vimPlugins.nvim-treesitter-parsers.matlab</li>
    <li>vimPlugins.nvim-treesitter-parsers.ocaml</li>
    <li>vimPlugins.nvim-treesitter-parsers.ocaml_interface</li>
    <li>vimPlugins.nvim-treesitter-parsers.robot</li>
    <li>vimPlugins.nvim-treesitter-parsers.tsx</li>
    <li>vimPlugins.nvim-treesitter-parsers.typescript</li>
    <li>vimPlugins.nvim-treesitter-parsers.wing</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.octo-nvim</li>
    <li>vimPlugins.onedarkpro-nvim</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.preto</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.rose-pine</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.starrynight</li>
    <li>vimPlugins.tabline-nvim</li>
    <li>vimPlugins.telescope-cheat-nvim</li>
    <li>vimPlugins.telescope-frecency-nvim</li>
    <li>vimPlugins.telescope-fzf-native-nvim</li>
    <li>vimPlugins.telescope-fzf-writer-nvim</li>
    <li>vimPlugins.telescope-fzy-native-nvim</li>
    <li>vimPlugins.telescope-media-files-nvim</li>
    <li>vimPlugins.telescope-nvim</li>
    <li>vimPlugins.telescope-symbols-nvim</li>
    <li>vimPlugins.telescope-undo-nvim</li>
    <li>vimPlugins.telescope-z-nvim</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.trouble-nvim</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vifm-vim</li>
    <li>vimPlugins.vim-just</li>
    <li>vimPlugins.vim-nix</li>
    <li>vimPlugins.vim-paper</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vimtex</li>
    <li>vimPlugins.wildfire-vim</li>
    <li>vimPlugins.yescapsquit-vim</li>
    <li>vimPlugins.zig-vim</li>
    <li>vimacs</li>
  </ul>
</details>

**Comment by figsoda** [comment]
I merged #245187 since this would run into merge conflicts

**Comment by lucasew** [comment]
Working on it

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 245275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vimPlugins.sg-nvim</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>vimPlugins.clangd_extensions-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.minimap-vim</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-pqf</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.fortran</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.preto</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.starrynight</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vim-paper</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vimtex</li>
    <li>vimPlugins.yescapsquit-vim</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 245275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>vimPlugins.clangd_extensions-nvim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.minimap-vim</li>
    <li>vimPlugins.neorg</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-pqf</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.fortran</li>
    <li>vimPlugins.nvim-treesitter-textobjects</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.preto</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.sg-nvim</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.starrynight</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vim-paper</li>
    <li>vimPlugins.vim-test</li>
    <li>vimPlugins.vimtex</li>
    <li>vimPlugins.yescapsquit-vim</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -1307,8 +1307,8 @@ final: prev:
     src = fetchFromGitHub {
       owner = "p00f";
       repo = "clangd_extensions.nvim";
-      rev = "b41da6b0838bd07d1d241f6bc09d054dc5f551fd";
-      sha256 = "04q642hjifrnhffb6vzg2km2vvw3fvghr9rh7l9s37ja4m3g2n3v";
+      rev = "4a822fca87c40099a4e8b27f396e7b24114b60d0";
+      sha256 = "1jw9hrb4577vn6yx037bd7diklcpkajasw7mv70mdwlxckdvvn02";
```
Comment (AndersonTorres) [inline pkgs/applications/editors/vim/plugins/generated.nix:1311]: Can the scripts be updated to use SRI hashes?

Not for now, but in a future PR.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (lucasew) [reply pkgs/applications/editors/vim/plugins/generated.nix:1311]: IDK

I only run the script

**Review by figsoda — APPROVED** [review]

**Pull request merged**